.cirrus.yml

```diff
@@ -992,7 +1022,17 @@ upgrade_test_task:
     name: "Upgrade test: from $PODMAN_UPGRADE_FROM"
     alias: upgrade_test
     # Docs: ./contrib/cirrus/CIModes.md
-    only_if: *not_tag_magic
+    # only when: - it is not a PR (we never want to skip on nightly tests); or
+    #            - CI:ALL title is set; or
+    #            - danger files are changed; or
+    #            - upgrade test code is changed; or
+    #            - source code is changed AND NOT only test files
+    only_if: >-
+        $CIRRUS_PR == '' ||
+        $CIRRUS_CHANGE_TITLE =~ '.*CI:ALL.*' ||
+        changesInclude('.cirrus.yml', 'Makefile', 'contrib/cirrus/**', 'vendor/**', 'hack/**', 'version/rawversion/*') ||
+        changesInclude('test/upgrade/**') ||
```

Comment:
This also needs `changesInclude('test/system/**')` because https://github.com/containers/podman/blob/b5bfd7233b1a3d500f0df1968dfcc2bd5c3c30ce/test/upgrade/helpers.bash#L3-L4

Reply:
Actually, this should be `changesInclude('test/system/*.bash')` -- `.bash` means "this is an include file", as opposed to `.bats` which means "this is a test". It would be stupid to run upgrade/farm tests if only a `system/*.bats` file changes.

Reply:
Ack, I try to keep the rules generic but I guess there is strong point made here that we do not want to many pointless runs when only system tests are touched.